atom/browser/api/atom_api_web_contents.cc

```diff
@@ -548,6 +559,58 @@ void WebContents::SetAllowTransparency(bool allow) {
   }
 }
 
+content::ServiceWorkerContext* WebContents::GetServiceWorkerContext() {
+  auto context = web_contents()->GetBrowserContext();
+  auto site_instance = web_contents()->GetSiteInstance();
+  if (!context && !site_instance)
+    return nullptr;
+
+  content::StoragePartition* storage_partition =
+    content::BrowserContext::GetStoragePartition(
+      context, site_instance);
+
+  DCHECK(storage_partition);
+
+  return storage_partition->GetServiceWorkerContext();
+}
+
+void WebContents::HasServiceWorker(mate::Arguments* args) {
+  GURL url;
+  base::Callback<void(bool)> callback;
+
+  if (!(args->Length() == 2 && args->GetNext(&url)
```

Comment:
You don't need to parse arguments manually, put them in the definition of `WebContents::HasServiceWorker` and conversion will be handled automatically.